tools/crm_mon.c

```diff
@@ -2199,17 +2199,19 @@ mon_refresh_display(gpointer user_data)
 #endif
 
         case mon_output_plain:
+            out->reset(out);
             print_status(out, mon_data_set, crm_errno2exit(history_rc), stonith_history,
                          options.mon_ops, show, options.neg_location_prefix,
                          options.only_node, options.only_rsc);
+            out->finish(out, CRM_EX_OK, true, NULL);
             break;
 
         case mon_output_unset:
         case mon_output_none:
             break;
     }
 
-    if (options.daemonize) {
+    if (options.daemonize && output_format != mon_output_plain) {
```

Comment:
This is a quirk of how the various formats work.  For text mode, output happens automatically (except for maybe having to flush the file handle).  The reset does an freopen to get the output file ready for writing on the next lap through daemonize mode.  But that'll also wipe out what was just printed.  So that's why there's extra hoops here for text output to jump through.

For the XML and HTML formatters, nothing gets written out until reset is called.  All the output is just batched up in xmlNodePtrs.  reset calls freopen to get the output file ready which wipes out the previous contents, and only after that happens does anything new get written to the file.  So that's why this is weird.

Reply:
I'm thinking reset() is broken if the caller needs format awareness. We could replace it with bookend calls, one that should be called before a new output block and one after.

Reply:
reset was supposed to be a convenience to not have to call finish and init.  This is the only place it's being used.  We could just nuke it and make sure finish+init acts the same way, and then just call those.  I'm not sure it's totally going to solve the problem without trying first.

Reply:
Well, doing so is going to require moving the block in pcmk__output_new that opens the destination into the ->init functions for each formatter.  I guess I could make it its own function too, but that seems annoying for such a trivial piece of code.  Having done that, it's probably then a simple matter of just making sure ->init and ->finish are called in the right places.

Reply:
Okay, I think I've got all the reset stuff sorted out.